tower-sessions-core/src/session.rs

```diff
@@ -857,29 +856,48 @@ impl Session {
 /// Id::default();
 /// ```
 #[derive(Copy, Clone, Debug, Deserialize, Serialize, Eq, Hash, PartialEq)]
-pub struct Id(pub Uuid); // TODO: By this being public, it may be possible to override UUIDv4,
-                         // which is undesirable.
+pub struct Id(pub i128); // TODO: By this being public, it may be possible to override the
```

Comment:
I didn't think of it at the time, but would `u128` be a better option here? For example, `Uuid` provides `from_128` which makes this change backwards compatible.

Maybe there was another reason for picking `i128` specifically?

Reply:
There’s no real befit between i128 and u128, however, UUID still has the issue of six “static” bits that are required to stay the same no matter what, at least in UUIDv4. Is it a _major_ security concern? Likely not, to be honest. But it’s also fairly trivial to avoid, imho, and worth it.

Reply:
I'm starting to wonder if `Id` should really just be `String`: while this means stores can't make assumptions about the size of the session key, it makes it possible to support IDs of arbitrary lengths.

The primary use case for this would be cookies as a storage layer, which is of course not allowed under the OWASP guidelines but seems to be something that virtually every session implementation supports (and for one reason or another is something people think they want).